src/template.ts

```diff
@@ -26,7 +26,7 @@ export function CodeTemplate(tempString: string, nodeKind?: number | number[]):
             let self = this;
             let retValue = target.apply(self, arguments);
             let [code, statements] = processTemplate(tempString, self);
-            if (statements)
+            if (statements && scope.statements)
```

Comment:
hmmm... why do we need this?

Reply:
Tests fail without this when DEBUG=1 (my new debug feautre, that backtraces on every template instantiation).

Reply:
it is extremly useful for template debugging IMO.

Reply:
Here is why this is needed: When emmiting top-level declarations it will use `code` (and not `statements`). There is no scope.statements there. However, with DEBUG=1, statements is set (by the debugging comments). So this prevents that from breaking anything.

Reply:
Yes, so the "debug logging feature" is directly interfering with the engine code. This is as strong code smell as I can imagine.

Have you never faced situations e.g. that when you enable logs, everything works, but as soon as you disable them, everything breaks?

Let's exclude this logging and it's side effects from the PR please.